tests/test_workflows/test_all/testAllHeadless.py

```diff
@@ -0,0 +1,123 @@
+
+
+class TestHeadlessWorkflowStartupProjectCreation(object):
+    """Start a headless shell and create a project for each workflow"""
+    @classmethod
+    def setupClass(cls):
+        cls.workflow_list = list(getAvailableWorkflows())
+
+    def test_workflow_creation_headless(self):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            for wf in self.workflow_list:
+                yield self.start_workflow_create_project_headless, wf, temp_dir
+
+    def start_workflow_create_project_headless(self, workflow_class_tuple, temp_dir):
+        """Tests project file creation via the command line
+        Args:
+            workflow_class_tuple (tuple): tuple returned from getAvailableWorkflows
+              with (workflow_class, workflow_name, workflow_class.workflowDisplayName)
+        """
+        workflow_class, workflow_name, display_name = workflow_class_tuple
+        logger.debug(f'starting {workflow_name}')
+        project_file = os.path.join(
+            temp_dir,
+            f'test_project_{"_".join(workflow_name.split())}.ilp'
+        )
+        args = [
+            '--headless',
+            f'--new_project={project_file}',
+            f'--workflow={workflow_name}',
+        ]
+        # Clear the existing commandline args so it looks like we're starting fresh.
+        sys.argv = ['ilastik.py']
+        sys.argv.extend(args)
+
+        # Start up the ilastik.py entry script as if we had launched it from the command line
+        parsed_args, workflow_cmdline_args = ilastik_main.parser.parse_known_args()
+
+        shell = ilastik_main.main(
+            parsed_args=parsed_args, workflow_cmdline_args=workflow_cmdline_args, init_logging=False)
+
+        shell.closeCurrentProject()
+
+        # now check if the project file has been created:
+        assert os.path.exists(project_file), f"Project File {project_file} creation not successful"
+
+    def test_workflow_loading_headless(self):
+        with tempfile.TemporaryDirectory() as temp_dir:
+            for wf in self.workflow_list:
+                yield self.start_workflow_load_project_headless, wf, temp_dir
+
+    def create_project_file(self, workflow_class, project_file_name):
+        newProjectFile = ProjectManager.createBlankProjectFile(project_file_name, workflow_class, [])
+        newProjectFile.close()
+
+    def start_workflow_load_project_headless(self, workflow_class_tuple, temp_dir):
+        """Tests opening project files in headless mode via the command line
+        Args:
+            workflow_class_tuple (tuple): tuple returned from getAvailableWorkflows
+              with (workflow_class, workflow_name, workflow_class.workflowDisplayName)
+        """
+        workflow_class, workflow_name, display_name = workflow_class_tuple
+        logger.debug(f'starting {workflow_name}')
+        project_file = os.path.join(
+            temp_dir,
+            f'test_project_{"_".join(workflow_name.split())}.ilp'
```

Comment:
To prevent mismatches between project filenames, it might be safer to create a method for the filename construction and use it here as well as above in the project creation.

Looks good otherwise!

Reply:
thanks :)